src/test/java/bi/deep/DistinctCountTimeseriesQueryTest.java

```diff
@@ -61,103 +66,193 @@ public void testTimeseriesWithDistinctCountAgg() throws Exception {
                         timestamp,
-                        Lists.newArrayList(visitor_id, client_type),
-                        ImmutableMap.of(visitor_id, "1", client_type, "iphone")
+                        Lists.newArrayList(VISITOR_ID, CLIENT_TYPE),
+                        ImmutableMap.of(VISITOR_ID, "1", CLIENT_TYPE, "iphone")
                 )
         );
         index.add(
                 new MapBasedInputRow(
                         timestamp,
-                        Lists.newArrayList(visitor_id, client_type),
-                        ImmutableMap.of(visitor_id, "2", client_type, "android")
+                        Lists.newArrayList(VISITOR_ID, CLIENT_TYPE),
+                        ImmutableMap.of(VISITOR_ID, "2", CLIENT_TYPE, "android")
                 )
         );
+    }
 
+    @Test
+    public void testFailQuery() {
         TimeseriesQuery queryToFail = Druids.newTimeseriesQueryBuilder()
                 .dataSource(QueryRunnerTestHelper.DATA_SOURCE)
                 .granularity(QueryRunnerTestHelper.ALL_GRAN)
                 .intervals(QueryRunnerTestHelper.FULL_ON_INTERVAL_SPEC)
                 .aggregators(
                         Lists.newArrayList(
                                 QueryRunnerTestHelper.ROWS_COUNT,
-                                new ExactDistinctCountAggregatorFactory("UV", visitor_id, 2, true)
+                                new ExactDistinctCountAggregatorFactory("UV", ImmutableList.of(VISITOR_ID), 2, true)
                         )
                 )
                 .build();
 
         Assert.assertThrows(RuntimeException.class, () -> engine.process(queryToFail, new IncrementalIndexStorageAdapter(index), new DefaultTimeseriesQueryMetrics()).toList());
+    }
 
-
+    @Test
+    public void testPartialQuery() {
         TimeseriesQuery partialQuery = Druids.newTimeseriesQueryBuilder()
                 .dataSource(QueryRunnerTestHelper.DATA_SOURCE)
                 .granularity(QueryRunnerTestHelper.ALL_GRAN)
                 .intervals(QueryRunnerTestHelper.FULL_ON_INTERVAL_SPEC)
                 .aggregators(
                         Lists.newArrayList(
                                 QueryRunnerTestHelper.ROWS_COUNT,
-                                new ExactDistinctCountAggregatorFactory("UV", visitor_id, 2, false)
+                                new ExactDistinctCountAggregatorFactory("UV", ImmutableList.of(VISITOR_ID), 2, false)
                         )
                 )
                 .build();
 
         final Iterable<Result<TimeseriesResultValue>> results =
                 engine.process(partialQuery, new IncrementalIndexStorageAdapter(index), new DefaultTimeseriesQueryMetrics()).toList();
-
-        HashSet<Integer> mutableSet = new HashSet<>();
-        mutableSet.add("0".hashCode());
-        mutableSet.add("1".hashCode());
+        Set<Integer> set = ImmutableSet.of(ImmutableList.of("0").hashCode(), ImmutableList.of("1").hashCode());
```

Comment:
Why do you need to create sublists inside of the set creation?

Reply:
The reason behind is that we are using List hashcode rather than individual object hashcode as there are multiple dimensions now.